atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -472,16 +474,19 @@ public Map<byte[], Iterator<Map.Entry<Cell, byte[]>>> getRowsColumnRangeIterator
                 keyValueService.getRowsColumnRange(tableRef, rows, columnRangeSelection, getStartTimestamp());
         ImmutableSortedMap.Builder<byte[], Iterator<Map.Entry<Cell, byte[]>>> postFilteredResultsBuilder =
                 ImmutableSortedMap.orderedBy(PtBytes.BYTES_COMPARATOR);
-        for (Map.Entry<byte[], RowColumnRangeIterator> e : rawResults.entrySet()) {
-            byte[] row = e.getKey();
-            RowColumnRangeIterator rawIterator = e.getValue();
-
-            // Sadly we can't close properly here without an ABI break :/
-            @SuppressWarnings("MustBeClosedChecker")
-            ClosableIterator<Map.Entry<Cell, byte[]>> postFilteredIterator =
-                    getPostFilteredColumns(tableRef, columnRangeSelection, row, rawIterator);
-
-            postFilteredResultsBuilder.put(row, scopeToTransaction(postFilteredIterator));
+        for (byte[] row : rows) {
+            Iterator<Entry<Cell, byte[]>> entryIterator = Optional.ofNullable(rawResults.get(row))
```

Comment:
Does `keyValueService.getRowsColumnRange` guarantee that the keys in the returned map are the same instance that was given in the `rows` parameter? If not, this `get` call is not going to work properly.

Reply:
from an API perspective I don't think there is anything guaranteeing that the keys are identical `byte[]` though in practice I believe they all are. In some cases the map returned `KeyValueService#getRowsColumnRange` is a `HashMap` (such as for CassandraKVS) and in others it is an `IdentityHashMap`. 

We could make this safer with something like though there's going to be overhead in many cases to construct and sort the map:
```
        ImmutableSortedMap<byte[], RowColumnRangeIterator> rawResults =
                ImmutableSortedMap.<byte[], RowColumnRangeIterator>orderedBy(PtBytes.BYTES_COMPARATOR)
                        .putAll(keyValueService.getRowsColumnRange(
                                tableRef, rows, columnRangeSelection, getStartTimestamp()))
                        .buildOrThrow();
```

https://github.com/palantir/atlasdb/blob/e6140c1ab94f5ec70ab775eeb3be4b4d72c12263/atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java#L882-L887


https://github.com/palantir/atlasdb/blob/e6140c1ab94f5ec70ab775eeb3be4b4d72c12263/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/KeyValueServices.java#L171-L192

Reply:
good flag - yeah in practice they should be but you're right that this is not guaranteed by the api

Reply:
(we've ended up going with the sorted-map approach @schlosna used in #6425)